contracts/starknet/teleport_oracle_auth.cairo

```diff
@@ -225,22 +320,22 @@ func request_mint{
 //     }
 func validate{
     syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, ecdsa_ptr: SignatureBuiltin*
-}(message: felt, sigs: Signature*, sigs_len: felt, threshold_: felt) {
+}(message: felt, signatures_len: felt, signatures: Signature*, threshold_: felt) {
     if (threshold_ == 0) {
         return ();
     }
 
-    assert_lt(0, sigs_len);
+    assert_lt(0, signatures_len);
 
-    let sig: Signature = sigs[0];
+    let sig: Signature = signatures[0];
 
     let (valid_sig) = _signers.read(sig.pk);
 
     assert valid_sig = 1;
 
     verify_ecdsa_signature(message, sig.pk, sig.r, sig.s);
```

Comment:
shouldn't we check somewhere that there is more signatures than the threshold?

Reply:
Both `signatures_len` and `threshold_` are decreased in each validate call. If initially `signatures_len` < `threshold_` `validate` will revert (line: 328). Otherwise `validate` will return without an error (line: 325).